terraform/modules/application/alarms.tf

```diff
@@ -38,6 +38,44 @@ resource "aws_cloudwatch_metric_alarm" "app_memory" {
   }
 }
 
+resource "aws_cloudwatch_metric_alarm" "app_tasks_exited" {
+  alarm_actions             = [var.sns_topic_arn]
+  alarm_name                = "${var.prefix}-app-tasks-exited"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  datapoints_to_alarm       = 1
+  evaluation_periods        = 1
+  metric_name               = "TriggeredRules"
+  namespace                 = "AWS/Events"
+  ok_actions                = [var.sns_topic_arn]
+  period                    = 8 * 60
+  statistic                 = "Sum"
+  threshold                 = 2 * var.app_task_desired_count
+  insufficient_data_actions = []
+
+  dimensions = {
+    RuleName = aws_cloudwatch_event_rule.app_task_exited.name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "app_service_action_problem" {
```

Comment:
I felt that that the conditions that would lead to a WARN or ERROR state were all things we'd want to know about straight away, rather than something we'd want to see happening over a period of time. That's why I went for 1/1/1 for datapoints/periods/threshold here.

The conditions that would lead to a WARN or ERROR state can be seen in the 'Service action events' section [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_cwe_events.html).

Reply:
P.S. I haven't tested this alarm works because I haven't managed to cause the event which would trigger it. I'm happy with the logic though.

Reply:
See my comment below about resource `arn` instead of `name` (may help)